core/sql/sqlcomp/CmpSeabaseDDLcommon.cpp

```diff
@@ -7764,52 +7734,103 @@ short CmpSeabaseDDL::initSeabaseAuthorization(
     // Add an error if none yet defined in the diags area
     if ( CmpCommon::diags()->getNumber(DgSqlCode::ERROR_) == 0)
       SEABASEDDL_INTERNAL_ERROR("initialize authorization");
-    cliRC = -1;
+
+    return -1;
   }
 
-  endXnIfStartedHere(cliInterface, xnWasStartedHere, cliRC);
+  // If DDL transactions are not enabled, commit the transaction so privmgr 
+  // schema exists in other processes
+  if (NOT ddlXns)
+  {
+    endXnIfStartedHere(cliInterface, xnWasStartedHere, 0);
+    if (beginXnIfNotInProgress(cliInterface, xnWasStartedHere))
+    {
+      SEABASEDDL_INTERNAL_ERROR("initialize authorization");
+      return -1;
+    }
+  }
 
-  return cliRC;
+  // change authorization status in compiler contexts
+  //CmpCommon::context()->setAuthorizationState (1);
```

Comment:
Yes, should remove this line.  Setting context is handled correctly in the setAuthStateInCmpContexts so this line is not needed.